types/store/index.ts

```diff
@@ -47,7 +47,7 @@ export type GlobalState = BaseGlobalState & {
             };
             focusedPostId: string;
             mobileView: boolean;
-            keepChannelIdAsUnread: Channel | null;
+            lastUnreadChannel: (Channel & {hadMentions: boolean}) | null; // Actually only an object with {id: string, hadMentions: boolean}
```

Comment:
I forget why the name of this field in the store was different than what we called it literally everywhere else, so I renamed it.

The ugly type is because I didn't want to go change this all throughout the channel selectors in mattermost-redux where it's used. We can do that cleanup once the old sidebar is removed and most of those selectors are removed.